src/Factory/AliasFactory.php

```diff
@@ -25,10 +27,11 @@ public static function create(User $user, ?string $localPart): Alias
         $alias->setDomain($domain);
         $alias->setDestination($user->getEmail());
         if (null === $localPart) {
-            $localPart = RandomStringGenerator::generate(24, false);
+            $localPart = RandomStringGenerator::generate(self::RANDOM_ALIAS_LENGTH, false);
+            $alias->setRandom();
         }
 
-        $alias->setSource($localPart.'@'.$domain->getName());
+        $alias->setSource(strtolower($localPart).'@'.$domain->getName());
```

Comment:
I think it would be better to do `strtolower()` in the `AliasHandler` and add a check for lowercase strings to `EmailValidator` (if it's not there yet). That way, `AliasFactory` stays stupid and doesn't have to care about string manipulation.